### PR TITLE
Implement snow within bounds indicators

### PR DESCRIPTION
**Pull request by huard**
### Pull Request Checklist:
- [x] This PR addresses an already opened issue (for bug fixes / features)
    - This PR relates to #604
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Documentation has been added / updated (for bug fixes / features)
- [x] HISTORY.rst has been updated (with summary of main changes)
- [ ] `bumpversion (major / minor / patch)` has been called on this branch
- [ ] Tags have been pushed (`git push --tags`)

* **What kind of change does this PR introduce?** 
Feature

* **Does this PR introduce a breaking change?** 
No

* **Other information**:
Please review indicator title, description, abstract for clarity.

### CODE REVIEW

File: xclim/testing/tests/test_precip.py [hunk 1]
```diff
@@ -371,6 +371,13 @@ def test_last_snowfall(self):
         )
 
 
+class TestDaysWithSnow:
+    def test_simple(self, prsn_series):
+        prsn = prsn_series(np.arange(366), start="1999-07-01")
+        out = atmos.days_with_snow(prsn, low="10 kg m-2 s-1", high="20 kg m-2 s-1")
+        assert out[0] == 11
```
Comment (aulemahal) [inline xclim/testing/tests/test_precip.py:378]: The "ERA5/daily_surface_cancities_1990-1993.nc" test dataset has prsn series if you want to test with real data.

**Review by aulemahal — APPROVED** [review]

**Comment by huard** [comment]
Question: Should we set the domain as ]low, high] instead of [low, high] to be able to partition the domain cleanly ?

File: xclim/testing/tests/test_precip.py (hunk 1 above)
Comment (huard) [reply xclim/testing/tests/test_precip.py:378]: Ok thanks, done.

**Comment by aulemahal** [comment]
Hum. maybe ]low, high] makes sense.  I think we had this discussion with @Zeitsperre some time ago.
It does not make a difference in most cases, as values are floats and are distributed "randomly". But for the situation where values are "digitized", that might be the expected result.

**Review by tlogan2000 — APPROVED** [review]

**Comment by tlogan2000** [comment]
> ]low, high] instead of [low, high]

?? lack of python skills must be showing ... what will this do differently?

**Comment by tlogan2000** [comment]
Never mind... I think i need a coffee ... go for ]low,high]

**Comment by Zeitsperre** [comment]
Yes, I recall this discussion. If we can go with ]low, high], it tackles a lot of issues that arise with model/reanalysis data.

**Pull request merged**